pkg/utils/utils.go

```diff
@@ -0,0 +1,68 @@
+package utils
```

Comment:
I consider creating a `utils` package to house random bits of logic an anti-pattern.  Once established it's very difficult to undo.  Can we figure out where these functions should really live before we start down the "junk drawer" path?

Reply:
moved IsExistingDir to Repos package
The others I moved to apply package for now but they are general purpose so should be somewhere that every package can import them from. ToJSON I duplicated in testutils package because it is used in the layer_test.go as well as apply.
Maybe importing them from my goutils wasn't that bad an idea. The flux team created a 'pkg' repo to hold general purpose utilities, maybe we should ask if we could have a fidelity/pkg repo?